apps/mwiniimporter/importer.cpp

```diff
@@ -834,35 +864,78 @@ void MwIniImporter::importGameFiles(multistrmap &cfg, const multistrmap &ini, co
 
     multistrmap::const_iterator it = ini.begin();
-    for(int i=0; it != ini.end(); i++) {
+    for (int i=0; it != ini.end(); i++)
+    {
         gameFile = baseGameFile;
         gameFile.append(this->numberToString(i));
 
         it = ini.find(gameFile);
-        if(it == ini.end()) {
+        if(it == ini.end())
             break;
-        }
 
-        for(std::vector<std::string>::const_iterator entry = it->second.begin(); entry!=it->second.end(); ++entry) {
+        for(std::vector<std::string>::const_iterator entry = it->second.begin(); entry!=it->second.end(); ++entry)
+        {
             std::string filetype(entry->substr(entry->length()-3));
             Misc::StringUtils::lowerCaseInPlace(filetype);
 
-            if(filetype.compare("esm") == 0 || filetype.compare("esp") == 0) {
+            if(filetype.compare("esm") == 0 || filetype.compare("esp") == 0)
+            {
                 boost::filesystem::path filepath(gameFilesDir);
                 filepath /= *entry;
-                contentFiles.push_back(std::make_pair(lastWriteTime(filepath, defaultTime), *entry));
+                contentFiles.push_back({lastWriteTime(filepath, defaultTime), filepath});
             }
         }
     }
 
     cfg.erase("content");
     cfg.insert( std::make_pair("content", std::vector<std::string>() ) );
 
-    // this will sort files by time order first, then alphabetical (maybe), I suspect non ASCII filenames will be stuffed.
+    // sort by timestamp
     sort(contentFiles.begin(), contentFiles.end());
-    for(std::vector<std::pair<std::time_t, std::string> >::const_iterator iter=contentFiles.begin(); iter!=contentFiles.end(); ++iter) {
-        cfg["content"].push_back(iter->second);
+
+    MwIniImporter::deplist unsorted_files;
+
+    ESM::ESMReader reader;
+    for (auto& file : contentFiles)
+    {
+        reader.open(file.second.string());
+        std::vector<std::string> deps;
+        for (auto& dep_file : reader.getGameFiles())
+        {
+            deps.push_back(dep_file.name);
+        }
+        unsorted_files.emplace_back(boost::filesystem::path(reader.getName()).filename().string(), deps);
+        reader.close();
     }
+
+    auto sorted_files = dependencySort(unsorted_files);
+
+    // hard-coded dependency Morrowind - Tribunal - Bloodmoon
+    {
+        auto find = [&sorted_files](const std::string& f)
+        {
+            return std::find_if(sorted_files.begin(), sorted_files.end(), [&f](std::string& file)
+            {
+                return Misc::StringUtils::ciEqual(file, f);
+            });
+        };
+
+        if(find("Morrowind.esm") != sorted_files.end())
+        {
+            auto foundTribunal = find("Tribunal.esm");
+            auto foundBloodmoon= find("Bloodmoon.esm");
+
+            if (foundBloodmoon != sorted_files.end() && foundTribunal != sorted_files.end())
+            {
+                auto s = *foundTribunal;
+                sorted_files.erase(foundTribunal);
```

Comment:
What happens if Tribunal appears before Bloodmoon? Would that invalidate the foundBloodmoon interator?